### PR TITLE
[backend-comparison] Refresh access token and display authenticated user name

### PR DESCRIPTION
**Pull request by syl20bnr**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

https://github.com/tracel-ai/burn/issues/1072

### Changes

Add automatic refresh of access token. Now both the access token and the refresh token are cached. The refresh is done by the user benchmark server.

A new endpoint `/users/me` in the user benchmark server returns the anonymized user name so that we can display it when executing `burnbench auth`. 

The `auth` command is now idempotent and can also refresh the access token so the CLI should now ask for the user's authorization only when strictly required.
 
### Testing

Tested with a local stack.

### CODE REVIEW

**Review by antimora — APPROVED** [review]
LGTM

File: backend-comparison/README.md [hunk 1]
```diff
@@ -96,14 +96,26 @@ authentication. To authenticate run the `auth` command, then follow the URL
...
 
 ```sh
-> cargo run --bin burnbench -- run auth
+> cargo run --bin burnbench -- auth
 ```
 
 If everything is fine you should get a confirmation in the terminal that your
 token has been saved to the burn cache directory.
 
+We don't store any of your personal information. A anonymized user name will
```
Comment (nathanielsimard) [inline backend-comparison/README.md:105]: `An anonymized`

File: backend-comparison/src/burnbenchapp/auth.rs [hunk 2]
```diff
@@ -1,46 +1,181 @@
...
-    let home_dir = dirs::home_dir().expect("an home directory should exist");
-    let path_dir = home_dir.join(".cache").join("burn").join("burnbench");
-    #[cfg(test)]
-    let path_dir = path_dir.join("test");
-    let path = Path::new(&path_dir);
-    path.join("token.txt")
+#[derive(Debug, Serialize, Deserialize)]
+pub(crate) struct Tokens {
+    /// Token returned once the Burnbench Github app has been authorized by the user.
+    /// This token is used to authenticate the user to the Burn benchmark serfver.
```
Comment (nathanielsimard) [inline backend-comparison/src/burnbenchapp/auth.rs:21]: serfver => server

File: backend-comparison/src/burnbenchapp/mod.rs [hunk 3]
```diff
@@ -12,3 +12,12 @@ use tui::TuiApplication as App;
...
 use term::TermApplication as App;
+
+const BENCHMARKS_TARGET_DIR: &str = "target/benchmarks";
+const USER_BENCHMARK_SERVER_URL: &str = if cfg!(debug_assertions) {
+    // development
+    "http://localhost:8000/"
+} else {
+    // production
+    "https://user-benchmark-server-gvtbw64teq-nn.a.run.app/"
+};
```
Comment (nathanielsimard) [inline backend-comparison/src/burnbenchapp/mod.rs:23]: We should add somewhere that running burnbench app without the --release flag won't update the results to the server.

**Review by nathanielsimard — CHANGES_REQUESTED** [review]
Minor changes

File: backend-comparison/src/burnbenchapp/mod.rs (hunk 3 above)
Comment (syl20bnr) [reply backend-comparison/src/burnbenchapp/mod.rs:23]: I updated the README and all the snippets to include the --release flag

**Comment by syl20bnr** [comment]
@nathanielsimard ready for another round.

**Pull request merged**